selectors/rhs.jsx

```diff
@@ -46,4 +47,15 @@ export const getSelectedPost = createSelector(
 export function makeGetCommentDraft(rootId) {
     const defaultValue = {message: '', fileInfos: [], uploadsInProgress: []};
     return makeGetGlobalItem(`comment_draft_${rootId}`, defaultValue);
-}
\ No newline at end of file
+}
+
+export const getPostsEmbedVisibleObj = createSelector(
+    (state) => state,
+    (state, {posts, previewCollapsed}) => ({posts, previewCollapsed}),
```

Comment:
Something important to note here is that this function returns a new object each time it's called, so it'll prevent memoization. Even if you were to do something like `(state, props) => (props)` here, props is likely changing each time this is called anyway, so it still wouldn't help. In the future though, you can break it up like this to have it work properly
```
createSelector(
    (state, {posts}) => posts,
    (state, {previewCollapsed}) => previewCollapsed,
    (posts, previewCollapsed) => {...}
);
```
As long as posts and previewCollapsed themselves don't change, it'll only be looking at their value to check if anything has changed instead of looking if the object containing them changed.

Reply:
Oh ya true. That's cool

Reply:
Yes made changes.